packages/diagrams/frontend/sirius-components-diagrams/src/renderer/handles/useConnectionCandidatesQuery.ts

```diff
@@ -25,7 +25,25 @@ import {
 } from '../connector/useConnector.types';
 import { NodeData } from '../DiagramRenderer.types';
 
+const ToolFields = gql`
+  fragment ToolFields on Tool {
+    __typename
+    ... on SingleClickOnTwoDiagramElementsTool {
+      candidates {
+        sources {
+          id
+        }
+        targets {
+          id
+        }
+      }
+      dialogDescriptionId
+    }
+  }
+`;
+
```

Comment:
You can just write the fragment below in the query, no need to parse it with gql

Reply:
ok